contributed_definitions/NXdispersive_material.nxdl.xml

```diff
@@ -0,0 +1,170 @@
+            </doc>
+        </field>
+        <field name="literature_dois" type="NX_CHAR" recommended="true">
+            <doc>
+                 This field holds the respective doi for the literature references.
+            </doc>
+        </field>
+        <field name="material_phase" type="NX_CHAR" optional="true">
+            <doc>
+                 The phase of the material
+            </doc>
+            <enumeration>
+                <item value="gas"/>
+                <item value="liquid"/>
+                <item value="solid"/>
+                <item value="other"/>
+            </enumeration>
+        </field>
+        <field name="material_phase_comment" type="NX_CHAR" optional="true">
+            <doc>
+                 Additional information about the phase if the
+                 material phase is other.
+            </doc>
+        </field>
+        <field name="additional_phase_information" type="NX_CHAR" recommended="true">
+            <doc>
+                 This field may be used to denote additional phase information,
+                 such as crystalin phase of a crystal (e.g. 4H or 6H for SiC) or
+                 if a measurement was done on a thin film or bulk material.
+            </doc>
+        </field>
+        <group name="dispersion_x" type="NXdispersion">
+            <doc>
+                 The dispersion along the optical axis of the material.
+                 This should be the only dispersion available for isotropic materials.
+                 For uniaxial materials this denotes the ordinary axis.
+                 For biaxial materials this denotes the x axis or epsilon 11 tensor element
+                 of the diagionalized permittivty tensor.
+            </doc>
+            <field name="model_name" type="NX_CHAR"/>
+            <group type="NXdispersion_table" recommended="true">
+                <field name="model_name" type="NX_CHAR"/>
+                <field name="convention" type="NX_CHAR"/>
+                <field name="wavelength" type="NX_CHAR"/>
+                <field name="dielectric_function" type="NX_CHAR" recommended="true"/>
+                <field name="refractive_index" type="NX_CHAR" recommended="true"/>
+            </group>
+            <group type="NXdispersion_function" recommended="true">
+                <field name="model_name" type="NX_CHAR"/>
+                <field name="formula" type="NX_CHAR"/>
+                <field name="convention" type="NX_CHAR"/>
+                <field name="energy_identifier" type="NX_CHAR" recommended="true"/>
+                <field name="wavelength_identifier" type="NX_CHAR" recommended="true"/>
+                <field name="representation" type="NX_CHAR"/>
+                <group type="NXdispersion_single_parameter">
+                    <field name="name" type="NX_CHAR"/>
+                    <field name="value" type="NX_CHAR"/>
+                </group>
+                <group type="NXdispersion_repeated_parameter">
```

Comment:
both single and repreated parameters are required. is it intentional?

Reply:
Yes, there may be a combination of both. For the case if only one of them are present it still should be set to an empty array. This way the software can always check for empty arrays but knows that the field will be present.